bears/vcs/git/GitCommitBear.py

```diff
@@ -151,6 +162,59 @@ def run(self,
             body,
             **self.get_issue_checks_metadata().filter_parameters(kwargs))
 
+    def check_revert(self, shortlog, body,
+                     allow_revert_commits_without_reason: bool = False):
+        """
+        Validates revert commits.
+
+        :param shortlog:
+            The shortlog message string.
+        :param body:
+            The body of the commit message of HEAD.
+        :param allow_revert_commits_without_reason:
+            Whether reason is needed for revert commits.
+        """
+        if not shortlog.startswith('Revert '):
+            return
+
+        shortlog_start_index = shortlog.find('"')
+        shortlog_end_index = shortlog.rfind('"')
+        reverted_shortlog = shortlog[shortlog_start_index +
+                                     1: shortlog_end_index]
+
+        body = body.strip('\n')
+        blankline = body.find('\n\n')
+        headline = body[:blankline] if blankline != -1 else body
+
+        if not headline.startswith('This reverts '):
+            yield Result(self, 'Invalid revert commit.')
+            return
+
+        sha_regex = re.compile(r'([0-9a-f]{40})')
+        commit_sha = re.findall(sha_regex, headline)[0]
+        command = ('git log -n 1 ' +
+                   str(commit_sha) + ' --pretty=%B')
+        previous_commit_stdout, previous_commit_stderr = run_shell_command(
+            command)
+
+        if previous_commit_stderr:
+            yield Result(self, 'Invalid sha for reverted commit.')
+            return
+
+        previous_commit_stdout = previous_commit_stdout.rstrip('\n')
+        position = previous_commit_stdout.find('\n')
+        old_shortlog = previous_commit_stdout[:position] if position != - \
+            1 else previous_commit_stdout
+
+        if old_shortlog != reverted_shortlog:
```

Comment:
This could be a problem if shortlog for the previous commit barely fit within the length limit as, to be a valid revert commit, it'll have to be rewritten with extra characters in the new commit and thereby cause an impossible situation.

I think the solution would be to rethink about the shortlog structure you require for a valid revert commit. I propose it should be written as
```
08b54a: Revert Commit

This reason to revert would be...

Reference if any
```

and this structure is also more aligned with the general guidelines.

Reply:
The format followed is one that is generated by git itself for revert commits. This PR only checks if a reason for revert is mentioned or not. See [link](https://github.com/coala/coala-bears/issues/337#issuecomment-372045150) 
I don't think changing shortlog structure would be most apt solution. One possible solution could be to increase shorlog length limit for revert commits (to maybe 60). @ksdme @jayvdb @damngamerz

Reply:
> One possible solution could be to increase shorlog length limit for revert commits

Yes, whenever a revert commit is detected, and revert commits have been permitted in the repo, other rules which might reject the commit must be turned off.